pennylane/templates/embeddings.py

```diff
@@ -295,6 +295,228 @@ def BasisEmbedding(features, wires):
     BasisState(features, wires=wires)
 
 
+def QAOAEmbedding(features, weights, wires, local_field='Y'):
+    r"""
+    Encodes :math:`N` features into :math:`n` qubits, using a layered, trainable quantum
+    circuit that is inspired by the QAOA ansatz.
+
+    A single layer applies two circuits or "Hamiltonians": The first encodes the features, and the second is
+    a variational ansatz inspired by a 1-dimensional Ising model. The feature-encoding circuit associates features with
+    the angles of :class:`RX` rotations. The Ising ansatz consists of trainable two-qubit ZZ interactions
+    :math:`e^{-i \alpha \sigma_z \otimes \sigma_z}`,
+    and trainable local fields :math:`e^{-i \frac{\beta}{2} \sigma_{\mu}}`, where :math:`\sigma_{\mu}`
+    can be chosen to be :math:`\sigma_{x}`, :math:`\sigma_{y}` or :math:`\sigma_{z}`
+    (default choice is :math:`\sigma_{y}` or the ``RY`` gate). :math:`\alpha, \beta` are adjustable gate parameters.
+
+    The number of features has to be smaller or equal to the number of qubits. If there are fewer features than
+    qubits, the feature-encoding rotation is replaced by a Hadamard gate.
+
+    This is an example for the full embedding circuit using 2 layers, 3 features, 4 wires, and ``RY`` local fields:
+
+    |
+
+    .. figure:: ../../_static/qaoa_layers.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    |
+
+    The argument ``weights`` contains an array of the :math:`\alpha, \beta` parameters for each layer.
+    The number of layers :math:`L` is derived from the first dimension of ``weights``.  If the embedding
+    acts on a single wire, ``weights`` has shape :math:`(L, )`, if the embedding acts on two wires, it has
+    shape :math:`(L, 3)`, and else it has shape :math:`(L, 2n)`
+
+    After the :math:`L`th layer,
```

Comment:
ReST is a bit weird, and the math won't render unless there is a space after the backtick

Reply:
(I didn't build this though, so could be wrong)

Reply:
True, it did swallowed some words!